scriptworker/test/test_config.py

```diff
@@ -74,6 +74,19 @@ def test_check_config_bad_keyring(t_config):
     assert "needs to start with %(gpg_home)s/" in "\n".join(messages)
 
 
+def test_check_config_invalid_ids(t_config):
```

Comment:
You could have used `pytest.mark.parametrize` to specify the key (e.g. `provisioner_id`) and the expected message (e.g., `provisioner_id doesn\'t match "^[a-zA-Z0-9-_]{1,22}$" (required by Taskcluster)`).  I don't think it's a big deal.

I do love that you added tests!